etstool.py

```diff
@@ -0,0 +1,334 @@
+------------------
+
+To change the packages installed during a test run, change the dependencies
+variable below.  To install a package from github, or one which is not yet
+available via EDM, add it to the `ci-src-requirements.txt` file (these will be
+installed by `pip`).
+
+Other changes to commands should be a straightforward change to the listed
+commands for each task. See the EDM documentation for more information about
+how to run commands within an EDM enviornment.
+
+"""
+
+import glob
+import os
+import shlex
+import subprocess
+import sys
+import time
+from timeit import default_timer
+from shutil import rmtree, copy as copyfile
+from tempfile import mkdtemp
+from contextlib import contextmanager
+
+import click
+
+# core dependencies that will be installed with edm
+dependencies = {
+    "enable",
+    "pyopengl",
+    "mayavi"}
+
+# test dependencies that will be installed with edm
+test_dependencies = {
+    "cython",
+    "nose",
+    "coverage",
+    "wheel"}
+
+# toolkit dependencies that will be installed with edm
+toolkits = {
+    'pyside': {'pyside'},
+    'pyqt4': {'pyqt<4.12'},  # FIXME: build of 4.12-1 appears to be bad
+}
+
+# latest dependencies that can be retrieved from pypi.
+pypi_dependencies = {
+    "traits",
+    "traitsui",
+    "pyopengl",
+    "mayavi"}
+
+# cutting edge dependencies that can be retrieved from source reposistories.
+repo_dependencies = {
+    "git+https://github.com/enthought/traits.git#egg=traits",
+    "git+https://github.com/enthought/traitsui.git#egg=traitsui",
+    "git+https://github.com/mcfletch/pyopengl.git#egg=pyopengl",
+    "git+https://github.com/nucleic/atom.git#egg=atom",
+    "git+https://github.com/enthought/mayavi.git#egg=mayavi"}
+
+environment_vars = {
+    'pyside': {'ETS_TOOLKIT': 'qt4', 'QT_API': 'pyside'},
+    'pyqt': {'ETS_TOOLKIT': 'qt4', 'QT_API': 'pyqt'}}
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.command()
+@click.option('--runtime', default='2.7', help='The python runtime version')
+@click.option(
+    '--toolkit', type=click.Choice(['pyside', 'pyqt4']),
+    default='pyside', help='The gui toolkit to use')
+@click.option(
+    '--environment', default=None, help='Override the default environment name')
+@click.option('--enaml', default='latest', help='The enaml version to build against')
+@click.option(
+    '--source', type=click.Choice(['edm', 'pypi', 'github']),
+    default='edm', help='The package source to use')
+def install(runtime, toolkit, environment, enaml, source):
+    """ Install project and dependencies into a clean EDM environment.
+
+    """
+    parameters = get_parameters(runtime, toolkit, environment, enaml, source)
+    # edm commands to setup the development environment
+    commands = [
+        "edm environments create {environment} --force --version={runtime}",
+        "edm install -y -e {environment} {test_packages} {edm_packages}",
+        "edm run -e {environment} -- pip install ."]
+    if source in ('pypi', 'github'):
```

Comment:
It might be safer to test the length of `parameters['pip_packages']` here instead, since there's more complicated logic happening in `get_parameters`.

Reply:
The base logic is that If source is not pypi or github we should not use pip to install dependencies no matter what is in the parameters dictionary. Checking parameters['pip_packages'] is an indirect check and I do not want  bug in the get_parameters to bubble up.

Reply:
I just wanted to point out that the logic around pip installation was in two places at opposite ends of the file. It allows the opportunity for mistakes in the future.

Reply:
The main reason why I prefer this setup is that if in the future a change in `get_parameters` adds packages to `pip_packages` by mistake. The `pip_package` check will incorrectly install packages in the environment and that mistake will not be detected unless we look at the logs with the aim to verify that the right packages from the right source have been installed in the logs. I have seen similar bug to stay undetected for months and that is why I prefer to prioritize  to be safe from such errors. 

In anycase, let me see what I can do.